File: android/src/main/java/com/skylabs/mixer/ResponseParameters.java
package com.skylabs.mixer;

public class ResponseParameters {
    // BaseResponse
    public static String status = "status";
    public static String message = "message";
    public static String data = "data";

    // MixerTimeResponse
    public static String milliSeconds = "milliSeconds";
    public static String seconds = "seconds";
    public static String minutes = "minutes";
    public static String hours = "hours";

    // PlaybackStateResponse
    public static String state = "state";

    // PlaybackStateBoolean
    public static String value = "value";

    // VolumeResponse
    public static String volume = "volume";

    // EqResponse
    public static String bassGain = "bassGain";
    public static String bassFrequency = "bassFrequency";
    public static String midGain = "midGain";
    public static String midFrequency = "midFrequency";
    public static String trebleGain = "trebleGain";
    public static String trebleFrequency = "trebleFrequency";

    // VolumeMeterResponse
    public static String meterLevel = "meterLevel";

    // InitResponse
//    public static String value = "value";

    // ChannelCountResponse
    public static String channelCount = "channelCount";
    public static String deviceName = "deviceName";

    // DestroyResponse
    public static String listenerName = "listenerName";
    public static String elapsedTimeEventName = "elapsedTimeEventName";

    // InitAudioSessionResponse
    public static String preferredInputPortType = "preferredInputPortType";
    public static String preferredInputPortName = "preferredInputPortName";
    public static String preferredIOBufferDuration = "preferredIOBufferDuration";
}

File: ios/Plugin/ResponseParameters.swift
//
//  ResponseParameters.swift
//  Plugin
//
//  Created by Skylabs Technology on 8/13/21.
//  Copyright © 2021 Max Lynch. All rights reserved.
//

import Foundation

/**
 * local constant mappings for response parameters
 */
public class ResponseParameters {
    // BaseResponse
    static var status: String = "status";
    static var message: String = "message";
    static var data: String = "data";

    // MixerTimeResponse
    static var milliSeconds: String = "milliSeconds";
    static var seconds: String = "seconds";
    static var minutes: String = "minutes";
    static var hours: String = "hours";

    // PlaybackStateResponse
    static var state: String = "state";

    // PlaybackStateBoolean
    static var value: String = "value";

    // VolumeResponse
    static var volume: String = "volume";

    // EqResponse
    static var bassGain: String = "bassGain";
    static var bassFrequency: String = "bassFrequency";
    static var midGain: String = "midGain";
    static var midFrequency: String = "midFrequency";
    static var trebleGain: String = "trebleGain";
    static var trebleFrequency: String = "trebleFrequency";

    // VolumeMeterResponse
    static var meterLevel: String = "meterLevel";

    // InitResponse
//    static var value: String = "value";

    // ChannelCountResponse
    static var channelCount: String = "channelCount";
    static var deviceName: String = "deviceName";

    // DestroyResponse
    static var listenerName: String = "listenerName";
    static var elapsedTimeEventName: String = "elapsedTimeEventName";

    // InitAudioSessionResponse
    static var preferredInputPortType: String = "preferredInputPortType";
    static var preferredInputPortName: String = "preferredInputPortName";
    static var preferredIOBufferDuration: String = "preferredIOBufferDuration";
    
    // FileValidationResponse
    static var isFileValid: String = "isFileValid";
    static var filePath: String = "filePath";

    // StateResponse
    // static var state: String = "state";
}
